pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -236,8 +234,26 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			status := OrchestrationStatus{Status: OrchestrationStatusCompleted, Message: fmt.Sprintf("failed to start osd provisioning on node %s", n.Name)}
 			if err := c.updateNodeStatus(n.Name, status); err != nil {
 				config.addError("failed to update node %s status. %+v", n.Name, err)
+				continue
 			}
 		}
+
+		if err := k8sutil.WaitForJobCompletion(c.context.Clientset, job, 300*time.Second); err != nil {
+			config.addError("failed to complete provision job for node %s. %+v", n.Name, err)
+			continue
+		}
+
+		// start the OSD pods, waiting for the provisioning to be completed
+		logger.Infof("start osds on node %s after provisioning is completed, if needed", n.Name)
+		c.completeProvision(config, n.Name)
```

Comment:
The pattern followed by the OSD creation is:
- Start the OSD prepare pod in parallel on all the nodes
- As the nodes are completed, initialize the OSD pods one-by-one in the `completeProvision()` method. 
So the OSDs are effectively already serialized for starting the OSD daemons. What is lacking right now is to wait inside `completeProvision()` for the PGs to stabilize after starting each OSD, which #2901 will fix. So I believe the remaining change with this PR is to only change `noout` and `norebalance` status if not already set, right?

Reply:
Running the OSD prepare pod in parallel is the main issue I was trying to solve.  This patch makes the provisioning of the prepare pod serial so only 1 node is upgraded at a time.  This is to ensure all replicas of a PG are not taken down at the same time and ensure running clients are not impacted by an upgrade.

Reply:
The running OSD daemons should not be affected by the prepare jobs running in parallel. The running daemons aren't upgraded until the `completeProvision()` step, and they are handled serially at that point. So I believe #2901 will solve your concern by waiting for clean PGs between each OSD daemon pod update.

Reply:
@travisn I have tested #2901 and it does solve the same issue I was trying to solve here.  So yes, all that is left is setting and unsetting the noout and norebalance flags.

Reply:
@mvollman thanks for trying out #2901, the only remaining issue is making it work with the CI...